dev/ColorPicker/InteractionTests/ColorPickerTests.cs

```diff
@@ -489,23 +489,60 @@ private void CanSelectColorFromSpectrum(ColorPickerTestSetupHelper setup, bool i
             // so we're bound to run into rounding errors along the way, which is what accounts for the slightly
             // different values we get in RTL where our x-axis is flipped.
-            setup.ExecuteAndWaitForColorChange(() => TapOnColorSpectrum(0.5, 0.5));
+            setup.ExecuteAndWaitForColorChange(() => TapOnColorSpectrumAndWaitForColorChange(0.5, 0.5));
             VerifySelectedColorIsNear(127, 255, 252);
             VerifySelectionEllipseIsNear(127, 127);
-            setup.ExecuteAndWaitForColorChange(() => TapOnColorSpectrum(0.25, 0.25));
+            setup.ExecuteAndWaitForColorChange(() => TapOnColorSpectrumAndWaitForColorChange(0.25, 0.25));
             VerifySelectedColorIsNear(163, isRTL ? 63 : 255, isRTL ? 255 : 63);
             VerifySelectionEllipseIsNear(isRTL ? 194 : 63, 63);
-            setup.ExecuteAndWaitForColorChange(() => TapOnColorSpectrum(0.75, 0.25));
+            setup.ExecuteAndWaitForColorChange(() => TapOnColorSpectrumAndWaitForColorChange(0.75, 0.25));
             VerifySelectedColorIsNear(151, isRTL ? 255 : 63, isRTL ? 63 : 255);
             VerifySelectionEllipseIsNear(isRTL ? 66 : 191, 63);
-            setup.ExecuteAndWaitForColorChange(() => TapOnColorSpectrum(0.25, 0.75));
+            setup.ExecuteAndWaitForColorChange(() => TapOnColorSpectrumAndWaitForColorChange(0.25, 0.75));
             VerifySelectedColorIsNear(224, isRTL ? 191 : 255, isRTL ? 255 : 191);
             VerifySelectionEllipseIsNear(isRTL ? 194 : 63, 192);
-            setup.ExecuteAndWaitForColorChange(() => TapOnColorSpectrum(0.75, 0.75));
+            setup.ExecuteAndWaitForColorChange(() => TapOnColorSpectrumAndWaitForColorChange(0.75, 0.75));
             VerifySelectedColorIsNear(220, isRTL ? 255 : 191, isRTL ? 191 : 255);
             VerifySelectionEllipseIsNear(isRTL ? 66 : 191, 192);
         }
 
+        [TestMethod]
+        public void VerifyHsvUpdatesOnSpectrumSelection()
+        {
+            using (var setup = SetupColorPickerTest())
+            {
+                TapOnColorSpectrumAndWaitForColorChange(0.5, 0.5);
+                VerifySelectedColorIsNear(127, 255, 252);
+                VerifySelectionEllipseIsNear(127, 127);
+               
+                WriteInTextBox(ValueTextBoxAutomationId, "0");
+                VerifySelectedColorIsNear(0, 0, 0);
+
+                // Since Value is 0, the RGB value stays black and does not trigger the colorChangedEvent,
+                // so this helper function is used instead.
+                TapOnColorSpectrum(0.25, 0.25);
+                Wait.ForIdle();
+
+                VerifySelectedColorIsNear(0, 0, 0);
+
+                // Allowing +/-1 devation
+                const double errorMargin = 1;
+                
+                Verify.IsLessThanOrEqual(90 - Double.Parse((new Edit(FindElement.ById(HueTextBoxAutomationId))).Value), errorMargin);
```

Comment:
Don't you need to take the absolute value of 90 - the value? With this code you only allow for 1 pixel under but infinite over.

Also, why do we expect deviation in the test?

Reply:
Fixed. 

The previous run failed from a 1pixel deviation, even though it was passing locally. I'm assuming it's a rendering difference from `TapOnColorSpectrum()` where it takes `colorSpectrum.BoundingRectangle.Width` to calculate it's x and yPosition.